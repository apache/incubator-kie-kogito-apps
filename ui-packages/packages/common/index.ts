export {
  default as DataTable
} from './src/components/Organisms/DataTable/DataTable';
export {
  default as KogitoSpinner
} from './src/components/Atoms/KogitoSpinner/KogitoSpinner';
export {
  default as PageToolbar
} from './src/components/Molecules/PageToolbar/PageToolbar';
export {
  default as KogitoPageLayout
} from './src/components/Templates/KogitoPageLayout/KogitoPageLayout';
export {
  default as AboutModalBox
} from './src/components/Molecules/AboutModalBox/AboutModalBox';
export {
  default as EndpointLink
} from './src/components/Atoms/EndpointLink/EndpointLink';
export {
  default as PageNotFound
} from './src/components/Molecules/PageNotFound/PageNotFound';
export {
  default as ServerUnavailable
} from './src/components/Molecules/ServerUnavailable/ServerUnavailable';
export { default as NoData } from './src/components/Molecules/NoData/NoData';
export {
  default as ServerErrors
} from './src/components/Molecules/ServerErrors/ServerErrors';
export {
  default as ProcessDescriptor
} from './src/components/Molecules/ProcessDescriptor/ProcessDescriptor';
export * from './src/components/Atoms/KogitoEmptyState/KogitoEmptyState';
export { default as LoadMore } from './src/components/Atoms/LoadMore/LoadMore';
export {
  default as DomainExplorer
} from './src/components/Organisms/DomainExplorer/DomainExplorer';
export {
  default as DomainExplorerListDomains
} from './src/components/Organisms/DomainExplorerListDomains/DomainExplorerListDomains';
export * from './src/utils/OuiaUtils';
export * from './src/graphql/types';
