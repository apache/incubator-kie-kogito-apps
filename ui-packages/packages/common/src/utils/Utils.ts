// tslint:disable: forin
// tslint:disable: no-floating-promises

const nestedCheck = (ele, valueObj) => {
  for (const key in ele) {
    const temp = ele[key];
    if (typeof temp[0] === 'object') {
      for (const nestedProp in temp[0]) {
        const nestedObj = {};
        const result = nestedCheck(temp[0], valueObj);
        if (valueObj.hasOwnProperty(nestedProp)) {
          valueObj[nestedProp] = result;
        } else {
          nestedObj[nestedProp] = result;
          valueObj = { ...valueObj, ...nestedObj };
        }
        return valueObj;
      }
    } else {
      const val = ele[key];
      const tempObj = {};
      tempObj[val[0]] = null;
      const firstKey = Object.keys(valueObj)[0];
      valueObj = { ...valueObj[firstKey], ...tempObj };
      return valueObj;
    }
  }
};

const checkFunc = (ele, valueObj) => {
  for (const key in ele) {
    const temp = ele[key];
    if (typeof temp[0] === 'object') {
      for (const nestedProp in temp[0]) {
        const nestedObj = {};
        if (valueObj.hasOwnProperty(nestedProp)) {
          const result = nestedCheck(temp[0], valueObj);
          valueObj[nestedProp] = result;
        } else {
          const result = checkFunc(temp[0], valueObj);
          nestedObj[nestedProp] = result;
          valueObj = { ...valueObj, ...nestedObj };
        }
        return valueObj;
      }
    } else {
      const val = ele[key];
      const tempObj = {};
      tempObj[val[0]] = null;
      valueObj = { ...valueObj, ...tempObj };
      return valueObj;
    }
  }
};
// function that validates null data
export const validateResponse = (obj, paramFields) => {
  let contentObj = {};
  for (const prop in obj) {
    const arr = [];
    if (obj[prop] === null) {
      const parentObj = {};
      paramFields.map(params => {
        if (params.hasOwnProperty(prop)) {
          arr.push(params);
        }
      });
      let valueObj = {};
      arr.forEach(ele => {
        valueObj = checkFunc(ele, valueObj);
      });
      parentObj[prop] = valueObj;
      contentObj = { ...contentObj, ...parentObj };
    } else {
      const elseObj = {};
      elseObj[prop] = obj[prop];
      contentObj = { ...contentObj, ...elseObj };
    }
  }
  return contentObj;
};

// function that frames object for query based on the selections
export const filterColumnSelection = (selectionArray, objValue) => {
  let res = {};
  if (selectionArray.length === 0) {
    res = objValue;
    return res;
  }
  for (let i = selectionArray.length - 1; i >= 0; i--) {
    if (i === selectionArray.length - 1) {
      if (selectionArray[i] === '-') {
        res = objValue;
      } else {
        res = { [selectionArray[i]]: [objValue] }; // assign the value
      }
    } else {
      res = { [selectionArray[i]]: [res] }; // put the prev object
    }
  }
  return res;
};

// function that removes single property from object
export const deleteKey = (testObj, pathArray) => {
  const _obj = testObj;
  const keys = pathArray;
  keys.reduce((acc, key, index) => {
    if (index === keys.length - 1) {
      delete acc[key];
      return true;
    }
    return acc[key];
  }, _obj);
  return _obj;
};
export const clearEmpties = obj => {
  for (const key in obj) {
    if (!obj[key] || typeof obj[key] !== 'object') {
      continue;
    }
    clearEmpties(obj[key]);
    if (Object.keys(obj[key]).length === 0) {
      delete obj[key];
    }
  }
  return obj;
};

// function adds new property to existing object
export const set = (obj, path, val) => {
  const keys = path.split(',');
  const lastKey = keys.pop();
  // tslint:disable-next-line: no-shadowed-variable
  const lastObj = keys.reduce(
    // tslint:disable-next-line: no-shadowed-variable
    (_obj, key) => (_obj[key] = obj[key] || {}),
    obj
  );
  lastObj[lastKey] = val;
};

// function removes duplicate objects inside array
export const removeDuplicates = (arr, comp) => {
  const unique = arr
    .map(e => e[comp])
    .map((e, i, final) => final.indexOf(e) === i && i)
    .filter(e => arr[e])
    .map(e => arr[e]);

  return unique;
};
