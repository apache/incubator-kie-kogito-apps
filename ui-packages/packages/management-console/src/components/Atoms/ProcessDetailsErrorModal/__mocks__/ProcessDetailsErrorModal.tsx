import React from 'react';

const ProcessDetailsErrorModal = () => {
  return <></>;
};

export default ProcessDetailsErrorModal;
