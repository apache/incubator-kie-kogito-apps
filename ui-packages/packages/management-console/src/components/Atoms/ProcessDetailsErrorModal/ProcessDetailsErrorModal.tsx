import React from 'react';
import {
  Modal,
  ModalVariant,
  ModalBoxBody,
  TextContent,
  Text
} from '@patternfly/react-core';
import { OUIAProps, componentOuiaProps } from '@kogito-apps/common';

interface IOwnProps {
  errorString: string;
  errorModalOpen: boolean;
  errorModalAction: JSX.Element[];
  handleErrorModal: () => void;
  label: string;
  title: JSX.Element;
}
const ProcessDetailsErrorModal: React.FC<IOwnProps & OUIAProps> = ({
  errorString,
  errorModalOpen,
  errorModalAction,
  handleErrorModal,
  label,
  title,
  ouiaId,
  ouiaSafe
}) => {
  const errorModalContent = (): JSX.Element => {
    return (
      <ModalBoxBody>
        <TextContent>
          <Text>{errorString}</Text>
        </TextContent>
      </ModalBoxBody>
    );
  };

  return (
    <Modal
      variant={ModalVariant.small}
      aria-labelledby={label}
      aria-label={label}
      title=""
      header={title}
      isOpen={errorModalOpen}
      onClose={handleErrorModal}
      actions={errorModalAction}
      {...componentOuiaProps(ouiaId, 'process-details-error-modal', ouiaSafe)}
    >
      {errorModalContent()}
    </Modal>
  );
};

export default ProcessDetailsErrorModal;
