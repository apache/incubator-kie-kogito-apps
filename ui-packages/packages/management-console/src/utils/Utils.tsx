import React from 'react';
import axios from 'axios';
import SVG from 'react-inlinesvg';
import {
  OnRunningIcon,
  CheckCircleIcon,
  BanIcon,
  PausedIcon,
  ErrorCircleOIcon,
  InfoCircleIcon,
  UndoIcon,
  ClockIcon
} from '@patternfly/react-icons';
import { GraphQL } from '@kogito-apps/common';
import ProcessInstanceState = GraphQL.ProcessInstanceState;
import ProcessInstance = GraphQL.ProcessInstance;
import JobStatus = GraphQL.JobStatus;
import {
  ProcessInstanceBulkList,
  OperationType
} from '../components/Molecules/ProcessListToolbar/ProcessListToolbar';
import { Title, TitleSizes } from '@patternfly/react-core';
import { JobsBulkList } from '../components/Templates/JobsManagementPage/JobsManagementPage';

export interface TriggerableNode {
  id: number;
  name: string;
  type: string;
  uniqueId: string;
  nodeDefinitionId: string;
}
/* tslint:disable:no-floating-promises */
export const ProcessInstanceIconCreator = (
  state: ProcessInstanceState
): JSX.Element => {
  switch (state) {
    case ProcessInstanceState.Active:
      return (
        <>
          <OnRunningIcon className="pf-u-mr-sm" />
          Active
        </>
      );
    case ProcessInstanceState.Completed:
      return (
        <>
          <CheckCircleIcon
            className="pf-u-mr-sm"
            color="var(--pf-global--success-color--100)"
          />
          Completed
        </>
      );
    case ProcessInstanceState.Aborted:
      return (
        <>
          <BanIcon className="pf-u-mr-sm" />
          Aborted
        </>
      );
    case ProcessInstanceState.Suspended:
      return (
        <>
          <PausedIcon className="pf-u-mr-sm" />
          Suspended
        </>
      );
    case ProcessInstanceState.Error:
      return (
        <>
          <ErrorCircleOIcon
            className="pf-u-mr-sm"
            color="var(--pf-global--danger-color--100)"
          />
          Error
        </>
      );
  }
};

export const JobsIconCreator = (state: JobStatus): JSX.Element => {
  switch (state) {
    case JobStatus.Error:
      return (
        <>
          <ErrorCircleOIcon
            className="pf-u-mr-sm"
            color="var(--pf-global--danger-color--100)"
          />
          Error
        </>
      );
    case JobStatus.Canceled:
      return (
        <>
          <BanIcon className="pf-u-mr-sm" />
          Canceled
        </>
      );
    case JobStatus.Executed:
      return (
        <>
          <CheckCircleIcon
            className="pf-u-mr-sm"
            color="var(--pf-global--success-color--100)"
          />
          Executed
        </>
      );
    case JobStatus.Retry:
      return (
        <>
          <UndoIcon className="pf-u-mr-sm" />
          Retry
        </>
      );
    case JobStatus.Scheduled:
      return (
        <>
          <ClockIcon className="pf-u-mr-sm" />
          Scheduled
        </>
      );
  }
};

export const setTitle = (
  titleStatus: string,
  titleText: string
): JSX.Element => {
  let icon = null;

  switch (titleStatus) {
    case 'success':
      icon = (
        <InfoCircleIcon
          className="pf-u-mr-sm"
          color="var(--pf-global--info-color--100)"
        />
      );
      break;
    case 'failure':
      icon = (
        <InfoCircleIcon
          className="pf-u-mr-sm"
          color="var(--pf-global--danger-color--100)"
        />
      );
      break;
  }

  return (
    <Title headingLevel="h1" size={TitleSizes['2xl']}>
      {icon}
      <span>{titleText}</span>
    </Title>
  );
};

export const handleSkip = async (
  processInstance: Pick<ProcessInstance, 'id' | 'processId' | 'serviceUrl'>,
  onSkipSuccess: () => void,
  onSkipFailure: (errorMessage: string) => void
) => {
  try {
    await axios.post(
      `${processInstance.serviceUrl}/management/processes/${processInstance.processId}/instances/${processInstance.id}/skip`
    );
    onSkipSuccess();
  } catch (error) {
    onSkipFailure(JSON.stringify(error.message));
  }
};

export const handleRetry = async (
  processInstance: Pick<ProcessInstance, 'id' | 'processId' | 'serviceUrl'>,
  onRetrySuccess: () => void,
  onRetryFailure: (errorMessage: string) => void
) => {
  try {
    await axios.post(
      `${processInstance.serviceUrl}/management/processes/${processInstance.processId}/instances/${processInstance.id}/retrigger`
    );
    onRetrySuccess();
  } catch (error) {
    onRetryFailure(JSON.stringify(error.message));
  }
};

export const handleAbort = async (
  processInstance: Pick<
    ProcessInstance,
    'id' | 'processId' | 'serviceUrl' | 'state'
  >,
  onRetrySuccess: () => void,
  onRetryFailure: (errorMessage: string) => void
) => {
  try {
    await axios.delete(
      `${processInstance.serviceUrl}/management/processes/${processInstance.processId}/instances/${processInstance.id}`
    );
    processInstance.state = ProcessInstanceState.Aborted;
    onRetrySuccess();
  } catch (error) {
    onRetryFailure(JSON.stringify(error.message));
  }
};

export const handleNodeInstanceRetrigger = (
  processInstance: Pick<ProcessInstance, 'id' | 'serviceUrl' | 'processId'>,
  node: Pick<GraphQL.NodeInstance, 'id'>,
  onRetriggerSuccess: () => void,
  onRetriggerFailure: (errorMessage: string) => void
) => {
  axios
    .post(
      `${processInstance.serviceUrl}/management/processes/${processInstance.processId}/instances/${processInstance.id}/nodeInstances/${node.id}`
    )
    .then(() => {
      onRetriggerSuccess();
    })
    .catch(error => {
      onRetriggerFailure(JSON.stringify(error.message));
    });
};

export const handleNodeInstanceCancel = (
  processInstance: Pick<ProcessInstance, 'id' | 'serviceUrl' | 'processId'>,
  node: Pick<GraphQL.NodeInstance, 'id'>,
  onCancelSuccess: () => void,
  onCancelFailure: (errorMessage: string) => void
) => {
  axios
    .delete(
      `${processInstance.serviceUrl}/management/processes/${processInstance.processId}/instances/${processInstance.id}/nodeInstances/${node.id}`
    )
    .then(() => {
      onCancelSuccess();
    })
    .catch(error => {
      onCancelFailure(JSON.stringify(error.message));
    });
};

export const performMultipleAction = async (
  instanceToBeActioned: ProcessInstanceBulkList,
  multiActionResult: (
    successInstances: ProcessInstanceBulkList,
    failedInstances: ProcessInstanceBulkList
  ) => void,
  processType: OperationType
) => {
  const successInstances = {};
  const failedInstances = {};
  for (const id of Object.keys(instanceToBeActioned)) {
    if (processType === OperationType.ABORT) {
      await handleAbort(
        instanceToBeActioned[id],
        () => {
          successInstances[id] = instanceToBeActioned[id];
        },
        errorMessage => {
          failedInstances[id] = instanceToBeActioned[id];
          failedInstances[id].errorMessage = errorMessage;
        }
      );
    } else if (processType === OperationType.SKIP) {
      await handleSkip(
        instanceToBeActioned[id],
        () => {
          successInstances[id] = instanceToBeActioned[id];
        },
        errorMessage => {
          failedInstances[id] = instanceToBeActioned[id];
          failedInstances[id].errorMessage = errorMessage;
        }
      );
    } else if (processType === OperationType.RETRY) {
      await handleRetry(
        instanceToBeActioned[id],
        () => {
          successInstances[id] = instanceToBeActioned[id];
        },
        errorMessage => {
          failedInstances[id] = instanceToBeActioned[id];
          failedInstances[id].errorMessage = errorMessage;
        }
      );
    }
  }
  multiActionResult(successInstances, failedInstances);
};

export const getProcessInstanceDescription = (
  processInstance: GraphQL.ProcessInstance
) => {
  return {
    id: processInstance.id,
    name: processInstance.processName,
    description: processInstance.businessKey
  };
};

export const getJobsDescription = (job: GraphQL.Job) => {
  return {
    id: job.id,
    name: job.processId
  };
};

// function containing Api call to update process variables
export const handleVariableUpdate = async (
  processInstance: Pick<ProcessInstance, 'id' | 'endpoint'>,
  updateJson: Record<string, unknown>,
  setDisplayLabel: (displayLabel: boolean) => void,
  setDisplaySuccess: (displaySuccess: boolean) => void,
  setUpdateJson: (updateJson: Record<string, unknown>) => void,
  setVariableError: (error: string) => void
): Promise<void> => {
  try {
    await axios
      .put(`${processInstance.endpoint}/${processInstance.id}`, updateJson)
      .then(res => {
        setUpdateJson(res.data);
        setDisplayLabel(false);
        setDisplaySuccess(true);
        setTimeout(() => {
          setDisplaySuccess(false);
        }, 2000);
      });
  } catch (error) {
    setVariableError(error.message);
  }
};

export const handleJobReschedule = async (
  job,
  repeatInterval,
  repeatLimit,
  rescheduleClicked,
  setErrorMessage,
  setRescheduleClicked,
  scheduleDate,
  refetch
): Promise<any> => {
  let parameter = {};
  if (repeatInterval === null && repeatLimit === null) {
    parameter = {
      expirationTime: new Date(scheduleDate)
    };
  } else {
    parameter = {
      expirationTime: new Date(scheduleDate),
      repeatInterval,
      repeatLimit
    };
  }
  try {
    await axios.patch(`${job.endpoint}/${job.id}`, parameter).then(res => {
      setRescheduleClicked(!rescheduleClicked);
      refetch();
    });
  } catch (error) {
    setRescheduleClicked(!rescheduleClicked);
    setErrorMessage(error.message);
    refetch();
  }
};

export const handleNodeTrigger = async (
  processInstance: Pick<ProcessInstance, 'id' | 'serviceUrl' | 'processId'>,
  node: Pick<TriggerableNode, 'nodeDefinitionId'>,
  onTriggerSuccess: () => void,
  onTriggerFailure: (error: string) => void
): Promise<void> => {
  try {
    await axios.post(
      `${processInstance.serviceUrl}/management/processes/${processInstance.processId}/instances/${processInstance.id}/nodes/${node.nodeDefinitionId}`
    );
    onTriggerSuccess();
  } catch (error) {
    onTriggerFailure(JSON.stringify(error.message));
  }
};

export const getTriggerableNodes = async (
  processInstance: Pick<GraphQL.ProcessInstance, 'processId' | 'serviceUrl'>
): Promise<TriggerableNode[]> => {
  try {
    const result = await axios.get(
      `${processInstance.serviceUrl}/management/processes/${processInstance.processId}/nodes`
    );
    return result.data;
  } catch (error) {
    return [];
  }
};

export const jobCancel = async (
  job: Pick<GraphQL.Job, 'id' | 'endpoint'>,
  setModalTitle: (title: JSX.Element) => void,
  setModalContent: (content: string) => void,
  refetch
) => {
  try {
    await axios.delete(`${job.endpoint}/${job.id}`);
    setModalTitle(setTitle('success', 'Job cancel'));
    setModalContent(`The job: ${job.id} is canceled successfully`);
    refetch();
  } catch (error) {
    setModalTitle(setTitle('failure', 'Job cancel'));
    setModalContent(
      `The job: ${job.id} failed to cancel. Error message: ${error.message}`
    );
    refetch();
  }
};

export const performMultipleCancel = async (
  jobsToBeActioned: (GraphQL.Job & { errorMessage?: string })[],
  multiActionResult: (
    successJobs: JobsBulkList,
    failedJobs: JobsBulkList
  ) => void
) => {
  const successJobs = {};
  const failedJobs = {};
  for (const job of jobsToBeActioned) {
    try {
      await axios.delete(`${job.endpoint}/${job.id}`);
      successJobs[job.id] = job;
    } catch (error) {
      job.errorMessage = error.errorMessage;
      failedJobs[job.id] = job;
      failedJobs[job.id].errorMessage = JSON.stringify(error.message);
    }
  }
  multiActionResult(successJobs, failedJobs);
};

export const getSvg = async (data, setSvg, setSvgError): Promise<void> => {
  setSvg(null);
  try {
    await axios
      .get(
        `/svg/processes/${data.ProcessInstances[0].processId}/instances/${data.ProcessInstances[0].id}`
      )
      .then(res => {
        const temp = <SVG src={res.data} />;
        setSvg(temp);
      });
  } catch (error) {
    if (error.response && error.response.status !== 404) {
      setSvgError(error.message);
    }
  }
};
