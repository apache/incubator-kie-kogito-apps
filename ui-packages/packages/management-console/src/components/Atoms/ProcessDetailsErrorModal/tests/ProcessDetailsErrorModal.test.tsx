import React from 'react';
import ProcessDetailsErrorModal from '../ProcessDetailsErrorModal';
import { getWrapper } from '@kogito-apps/common';
import { setTitle } from '../../../../utils/Utils';
import { Button } from '@patternfly/react-core';

const MockedIcon = (): React.ReactElement => {
  return <></>;
};
jest.mock('@patternfly/react-icons', () => ({
  ...jest.requireActual('@patternfly/react-icons'),
  InfoCircleIcon: () => {
    return <MockedIcon />;
  }
}));

const errorModalAction: JSX.Element[] = [
  <Button key="confirm-selection" variant="primary">
    OK
  </Button>
];
describe('Process details error modal tests', () => {
  const props = {
    errorString: '404-not found',
    errorModalOpen: true,
    errorModalAction: errorModalAction,
    handleErrorModal: jest.fn(),
    label: 'Error modal',
    title: setTitle('failure', 'Process Visualization')
  };

  it('Snapshot test with default props', () => {
    const wrapper = getWrapper(
      <ProcessDetailsErrorModal {...props} />,
      'ProcessDetailsErrorModal'
    );
    expect(wrapper).toMatchSnapshot();
    expect(
      wrapper
        .find('Text')
        .children()
        .text()
    ).toEqual('404-not found');
  });
});
