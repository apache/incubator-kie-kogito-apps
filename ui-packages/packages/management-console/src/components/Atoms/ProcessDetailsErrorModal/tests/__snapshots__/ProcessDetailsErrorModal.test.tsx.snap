// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Process details error modal tests Snapshot test with default props 1`] = `
<ProcessDetailsErrorModal
  errorModalAction={
    Array [
      <Button
        variant="primary"
      >
        OK
      </Button>,
    ]
  }
  errorModalOpen={true}
  errorString="404-not found"
  handleErrorModal={[MockFunction]}
  label="Error modal"
  title={
    <Title
      headingLevel="h1"
      size="2xl"
    >
      <InfoCircleIcon
        className="pf-u-mr-sm"
        color="var(--pf-global--danger-color--100)"
      />
      <span>
        Process Visualization
      </span>
    </Title>
  }
>
  <Modal
    actions={
      Array [
        <Button
          variant="primary"
        >
          OK
        </Button>,
      ]
    }
    appendTo={
      <body
        class="pf-c-backdrop__open"
      >
        <div>
          <div
            class="pf-c-backdrop"
          >
            <div
              class="pf-l-bullseye"
            >
              <div
                aria-describedby="pf-modal-part-2"
                aria-label="Error modal"
                aria-labelledby="pf-modal-part-0 Error modal"
                aria-modal="true"
                class="pf-c-modal-box pf-m-sm"
                data-ouia-component-id="0"
                data-ouia-component-type="PF4/ModalContent"
                data-ouia-safe="true"
                id="pf-modal-part-0"
                role="dialog"
              >
                <button
                  aria-disabled="false"
                  aria-label="Close"
                  class="pf-c-button pf-m-plain"
                  data-ouia-component-id="1"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 352 512"
                    width="1em"
                  >
                    <path
                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                      transform=""
                    />
                  </svg>
                </button>
                <header
                  class="pf-c-modal-box__header"
                >
                  <h1
                    class="pf-c-title pf-m-2xl"
                  >
                    <span>
                      Process Visualization
                    </span>
                  </h1>
                </header>
                <div
                  class="pf-c-modal-box__body"
                  data-ouia-component-type="process-details-error-modal"
                  data-ouia-safe="true"
                  id="pf-modal-part-2"
                >
                  <div
                    class="pf-c-modal-box__body"
                  >
                    <div
                      class="pf-c-content"
                    >
                      <p
                        class=""
                        data-pf-content="true"
                      >
                        404-not found
                      </p>
                    </div>
                  </div>
                </div>
                <footer
                  class="pf-c-modal-box__footer"
                >
                  <button
                    aria-disabled="false"
                    class="pf-c-button pf-m-primary"
                    data-ouia-component-id="2"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    type="button"
                  >
                    OK
                  </button>
                </footer>
              </div>
            </div>
          </div>
        </div>
      </body>
    }
    aria-describedby=""
    aria-label="Error modal"
    aria-labelledby="Error modal"
    className=""
    data-ouia-component-type="process-details-error-modal"
    data-ouia-safe={true}
    hasNoBodyWrapper={false}
    header={
      <Title
        headingLevel="h1"
        size="2xl"
      >
        <InfoCircleIcon
          className="pf-u-mr-sm"
          color="var(--pf-global--danger-color--100)"
        />
        <span>
          Process Visualization
        </span>
      </Title>
    }
    isOpen={true}
    onClose={[MockFunction]}
    ouiaSafe={true}
    showClose={true}
    title=""
    variant="small"
  >
    <Portal
      containerInfo={
        <div>
          <div
            class="pf-c-backdrop"
          >
            <div
              class="pf-l-bullseye"
            >
              <div
                aria-describedby="pf-modal-part-2"
                aria-label="Error modal"
                aria-labelledby="pf-modal-part-0 Error modal"
                aria-modal="true"
                class="pf-c-modal-box pf-m-sm"
                data-ouia-component-id="0"
                data-ouia-component-type="PF4/ModalContent"
                data-ouia-safe="true"
                id="pf-modal-part-0"
                role="dialog"
              >
                <button
                  aria-disabled="false"
                  aria-label="Close"
                  class="pf-c-button pf-m-plain"
                  data-ouia-component-id="1"
                  data-ouia-component-type="PF4/Button"
                  data-ouia-safe="true"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 352 512"
                    width="1em"
                  >
                    <path
                      d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                      transform=""
                    />
                  </svg>
                </button>
                <header
                  class="pf-c-modal-box__header"
                >
                  <h1
                    class="pf-c-title pf-m-2xl"
                  >
                    <span>
                      Process Visualization
                    </span>
                  </h1>
                </header>
                <div
                  class="pf-c-modal-box__body"
                  data-ouia-component-type="process-details-error-modal"
                  data-ouia-safe="true"
                  id="pf-modal-part-2"
                >
                  <div
                    class="pf-c-modal-box__body"
                  >
                    <div
                      class="pf-c-content"
                    >
                      <p
                        class=""
                        data-pf-content="true"
                      >
                        404-not found
                      </p>
                    </div>
                  </div>
                </div>
                <footer
                  class="pf-c-modal-box__footer"
                >
                  <button
                    aria-disabled="false"
                    class="pf-c-button pf-m-primary"
                    data-ouia-component-id="2"
                    data-ouia-component-type="PF4/Button"
                    data-ouia-safe="true"
                    type="button"
                  >
                    OK
                  </button>
                </footer>
              </div>
            </div>
          </div>
        </div>
      }
    >
      <ModalContent
        actions={
          Array [
            <Button
              variant="primary"
            >
              OK
            </Button>,
          ]
        }
        aria-describedby=""
        aria-label="Error modal"
        aria-labelledby="Error modal"
        boxId="pf-modal-part-0"
        className=""
        data-ouia-component-type="process-details-error-modal"
        data-ouia-safe={true}
        descriptorId="pf-modal-part-2"
        hasNoBodyWrapper={false}
        header={
          <Title
            headingLevel="h1"
            size="2xl"
          >
            <InfoCircleIcon
              className="pf-u-mr-sm"
              color="var(--pf-global--danger-color--100)"
            />
            <span>
              Process Visualization
            </span>
          </Title>
        }
        isOpen={true}
        labelId="pf-modal-part-1"
        onClose={[MockFunction]}
        ouiaSafe={true}
        showClose={true}
        title=""
        variant="small"
      >
        <Backdrop>
          <div
            className="pf-c-backdrop"
          >
            <FocusTrap
              active={true}
              className="pf-l-bullseye"
              focusTrapOptions={
                Object {
                  "clickOutsideDeactivates": true,
                }
              }
              paused={false}
            >
              <div
                className="pf-l-bullseye"
              >
                <ModalBox
                  aria-describedby="pf-modal-part-2"
                  aria-label="Error modal"
                  aria-labelledby="pf-modal-part-0 Error modal"
                  className=""
                  data-ouia-component-id={0}
                  data-ouia-component-type="PF4/ModalContent"
                  data-ouia-safe={true}
                  id="pf-modal-part-0"
                  style={Object {}}
                  variant="small"
                >
                  <div
                    aria-describedby="pf-modal-part-2"
                    aria-label="Error modal"
                    aria-labelledby="pf-modal-part-0 Error modal"
                    aria-modal="true"
                    className="pf-c-modal-box pf-m-sm"
                    data-ouia-component-id={0}
                    data-ouia-component-type="PF4/ModalContent"
                    data-ouia-safe={true}
                    id="pf-modal-part-0"
                    role="dialog"
                    style={Object {}}
                  >
                    <ModalBoxCloseButton
                      onClose={[MockFunction]}
                    >
                      <Button
                        aria-label="Close"
                        className=""
                        onClick={[MockFunction]}
                        variant="plain"
                      >
                        <button
                          aria-disabled={false}
                          aria-label="Close"
                          className="pf-c-button pf-m-plain"
                          data-ouia-component-id={1}
                          data-ouia-component-type="PF4/Button"
                          data-ouia-safe={true}
                          disabled={false}
                          onClick={[MockFunction]}
                          type="button"
                        >
                          <TimesIcon
                            color="currentColor"
                            noVerticalAlign={false}
                            size="sm"
                          >
                            <svg
                              aria-hidden={true}
                              aria-labelledby={null}
                              fill="currentColor"
                              height="1em"
                              role="img"
                              style={
                                Object {
                                  "verticalAlign": "-0.125em",
                                }
                              }
                              viewBox="0 0 352 512"
                              width="1em"
                            >
                              <path
                                d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                                transform=""
                              />
                            </svg>
                          </TimesIcon>
                        </button>
                      </Button>
                    </ModalBoxCloseButton>
                    <ModalBoxHeader>
                      <header
                        className="pf-c-modal-box__header"
                      >
                        <Title
                          headingLevel="h1"
                          size="2xl"
                        >
                          <h1
                            className="pf-c-title pf-m-2xl"
                          >
                            <InfoCircleIcon
                              className="pf-u-mr-sm"
                              color="var(--pf-global--danger-color--100)"
                            >
                              <MockedIcon />
                            </InfoCircleIcon>
                            <span>
                              Process Visualization
                            </span>
                          </h1>
                        </Title>
                      </header>
                    </ModalBoxHeader>
                    <ModalBoxBody
                      data-ouia-component-type="process-details-error-modal"
                      data-ouia-safe={true}
                      id="pf-modal-part-2"
                    >
                      <div
                        className="pf-c-modal-box__body"
                        data-ouia-component-type="process-details-error-modal"
                        data-ouia-safe={true}
                        id="pf-modal-part-2"
                      >
                        <ModalBoxBody>
                          <div
                            className="pf-c-modal-box__body"
                          >
                            <TextContent>
                              <div
                                className="pf-c-content"
                              >
                                <Text>
                                  <p
                                    className=""
                                    data-pf-content={true}
                                  >
                                    404-not found
                                  </p>
                                </Text>
                              </div>
                            </TextContent>
                          </div>
                        </ModalBoxBody>
                      </div>
                    </ModalBoxBody>
                    <ModalBoxFooter>
                      <footer
                        className="pf-c-modal-box__footer"
                      >
                        <Button
                          key="confirm-selection"
                          variant="primary"
                        >
                          <button
                            aria-disabled={false}
                            aria-label={null}
                            className="pf-c-button pf-m-primary"
                            data-ouia-component-id={2}
                            data-ouia-component-type="PF4/Button"
                            data-ouia-safe={true}
                            disabled={false}
                            type="button"
                          >
                            OK
                          </button>
                        </Button>
                      </footer>
                    </ModalBoxFooter>
                  </div>
                </ModalBox>
              </div>
            </FocusTrap>
          </div>
        </Backdrop>
      </ModalContent>
    </Portal>
  </Modal>
</ProcessDetailsErrorModal>
`;
